src/backend/linalg_internal_cpu/Det_internal.cpp

```diff
@@ -27,8 +31,8 @@ namespace cytnx {
         "[ERROR][Det_internal] Error in Lapack function 'zgetrf': Lapack INFO = ", info);
       od[0] = 1;
       bool neg = 0;
-      for (lapack_int i = 0; i < N; i++) {
-        od[0] *= (cytnx_complex128)_Rin[i * N + i];
+      for (int i = 0; i < N; i++) {
```

Comment:
the reason for having  lapack_int because of backend agnostic issue.  Here is to make it compatible to when the lapack/mkl is using 64bit backend/api v.s. 32bit. by replacing it with int you are actually limited the scope

Reply:
Updated